### PR TITLE
Collection method

### PR DESCRIPTION
**Pull request by AlessandroMinali**
Allow collection of objects returned by ORM to be surrealized:
https://github.com/nesaulov/surrealist/issues/13
https://github.com/nesaulov/surrealist/issues/12

Api: `Surrealist.surrealize(collection, camelize, include_root)`

Tested on ActiveRecord, Sequel, ROM, and Datamapper.

Should work for any ORM that provides a array of objects that we can `include Surrealist` and define `#json schema`.

### CODE REVIEW

**Comment by nesaulov** [comment]
@AlessandroMinali thanks! This looks right, but I think we have to have a lot more test for this feature. We need to be sure that `#surrealize_collection` works well with every ORM for:
* Inheritance
* All possible optional arguments:
  * `camelize`
  * `include_root`
  * `include_namespaces` (#14, WIP, I'll introduce PR soon)
  * `namespaces_nesting_level` (#14, WIP, I'll introduce PR soon)
* Nested objects inside DB, and other features like `has_many` and so on

Also we need to test all the exceptions, maybe have a `build_schema` for collection (I don't know if it's useful), and I'm quite sure I've missed something here. I will focus on this feature right after I'll close #14, so we can both think through various cases.

**Comment by AlessandroMinali** [comment]
@nesaulov https://github.com/nesaulov/surrealist/pull/20/commits/2cd217cdb2eed250405821ebe7c5faab44b2e3ba

Contains some spec POC for active record concerning the first two points (inheritance and params). I assume by inheritance you meant [Single table inheritance](http://api.rubyonrails.org/classes/ActiveRecord/Inheritance.html). If this look good I can start taking it to the other ORMs.

For the nested objects I think I have a naive way to do it, but blocking infinite recursion might be complicated. Also unsure how consistent behaviour is across ORMs. I'm leaving that to the end...or even better a separate PR since that's a pretty big changes/feature.

**Comment by nesaulov** [comment]
@AlessandroMinali this looks fine to me. But I suggest creating a separate spec folder, `spec/orms/` and have a separate file for each ORM, because I find current specs a bit hard to read. Important things to cover here apart from what I've listed earlier are associations and scopes, I think.
Also I've just realized that we need to test stuff like `joins`, `includes` and so on. Brr, a lot of stuff. Maybe it would be better if we focus only on AR in scope of this PR?
P.S. JRuby seems to have troubles with sqlite, I'll dig into that.

**Comment by AlessandroMinali** [comment]
@nesaulov https://github.com/nesaulov/surrealist/pull/20/commits/2450ba654e84ded2335b96012fdd9894910b0232

Separated out concerns in specs to make them a little cleaner to read through. Focusing on AR: added association and scope tests.

File: lib/surrealist.rb [hunk 1]
```diff
@@ -26,6 +26,39 @@ def included(base)
...
+      end
+
+      collection.map do |record|
+        record.surrealize(
+          camelize: camelize,
+          include_root: include_root,
+          include_namespaces: include_namespaces,
+          namespaces_nesting_level: namespaces_nesting_level
+        )
+      end
```
Comment (nesaulov) [inline lib/surrealist.rb]: I see a problem here: `#surrealize_collection` returns array now. Whereas it should return plain JSON string:
```ruby
> Surrealist.surrealize_collection(_)
=> ["{\"name\":\"rsbzaufj\"}", "{\"name\":\"xgjzrdfc\"}", "{\"name\":\"vjboywrp\"}"]
# should be
=> "[{\"name\":\"rsbzaufj\"},{\"name\":\"xgjzrdfc\"},{\"name\":\"vjboywrp\"}]"
```

Comment (nesaulov) [reply lib/surrealist.rb]: I guess solution would be
```ruby
JSON.dump(
        collection.map do |record|
          record.build_schema(
            camelize:                 camelize,
            include_root:             include_root,
            include_namespaces:       include_namespaces,
            namespaces_nesting_level: namespaces_nesting_level
          )
        end
      )
```

**Comment by AlessandroMinali** [comment]
@nesaulov Fixed and added simple join / includes spec.

I am unsure about what is needed for the jruby case or what it requires as I have no experience with it, maybe this can help you: https://stackoverflow.com/questions/7822623/rails-how-can-my-app-tell-if-it-is-running-in-mri-or-jruby

**Comment by nesaulov** [comment]
@AlessandroMinali yeah, I guess this is what it takes for JRuby to work. Could you please merge master (I've added env variable that allows C extensions) and update Gemfile according to SO answer?

**Comment by AlessandroMinali** [comment]
@nesaulov I have wasted a number of hours trying to get to jruby to work...I have no idea what I'm doing here and it seems like the adapters are outdated, relying on old versions of AR...

I had even tried setting up the mysql adapter to no avail. I'm not going to personally waste any more time here, so I'll leave it as an exercise for you to finish if you want it.

**Comment by nesaulov** [comment]
@AlessandroMinali sorry for your time, that's unfortunate. I guess we can just drop official JRuby support (remove it from CI), since the only problem is with tests.

**Comment by nesaulov** [comment]
@AlessandroMinali I've officially dropped JRuby support, so please merge master and build should pass.

**Comment by AlessandroMinali** [comment]
@nesaulov ready for merge

**Pull request merged**